src/components/map/LocationOverview.tsx

```diff
@@ -44,8 +46,24 @@ const mapToMapMarkers = (pointsOfInterest: any): MapMarker[] | undefined => {
 
         if (!latitude || !longitude) return undefined;
 
+        let icon = location;
+
+        if (item.category?.iconName) {
+          // remove location and locationIconAnchor from Icons for proper type perspective to all
+          // the other icons
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          const { location: _l, locationIconAnchor: _lia, ...OtherIcons } = Icons;
+          const iconName = item.category.iconName as keyof typeof OtherIcons;
+
+          // only use the icon if it is available from our icons
+          if (Object.keys(OtherIcons).includes(iconName)) {
+            icon = OtherIcons[iconName] as (color: any) => string;
+          }
+        }
```

Comment:
i was able to secure and type everything properly. this way we can ensure, that only available icons will be used and bad icon name strings will not result in errors. it will fallback to `location` icon.

<img width="660" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/1942953/e8d5a3a1-db35-4bf3-8faf-547aa03be308">

Reply:
Cool. Thank you 🎉